solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java

```diff
@@ -1167,8 +1191,38 @@ public RouteException(ErrorCode errorCode, NamedList<Throwable> throwables, Map<
     }
 
     LBSolrClient.Req req = new LBSolrClient.Req(request, theUrlList);
-    LBSolrClient.Rsp rsp = getLbClient().request(req);
-    return rsp.getResponse();
+    if (isAsyncRequest) {
+      CompletableFuture<LBSolrClient.Rsp> lbFuture = ((LBHttp2SolrClient) getLbClient()).requestAsync(req);
```

Comment:
Maybe we want to document somewhere that in order to support async requests, the internal lb client needs to extend `LBHttp2SolrClient`?